### PR TITLE
Configure accounts on cte dev

### PR DESCRIPTION
**Pull request by karenc**
- :bug: Correct zope domain in cte-dev group vars
 - Change authoring and publishing app url to use https and remove port
 - Add encrypted accounts secrets to cte-dev/group_vars

   Example usage:
    
    ```
    ansible-playbook -i environments/cte-dev/inventory main.yml --ask-become-pass --ask-vault-pass
    ```

 - :bug: Always restart supervisor to restart archive, publishing and authoring

    Add 5 seconds between the stop and start command in attempt to fix this
    error:
    
    ```
    TASK [configure authoring under supervisor]
    ************************************
    ok: [cte-cnx-dev.cnx.org]
    
    TASK [restart supervisor]
    ******************************************************
    fatal: [cte-cnx-dev.cnx.org]: FAILED! => {"changed": false, "failed":
    true, "msg": "Stopping supervisor: supervisord.\nStarting supervisor: "}
    ```

 - :sparkles: Add authoring to nginx config

 - Update webview settings.js to not use port numbers for backend servers

 - Use webview master on cte-dev

   Because webview production branch doesn't use authoring and publishing.

 - Install openstax-accounts when installing authoring

 - Hide authoring behind frontend domain using nginx

 - Change webview archive settings to use arclishing_domain

 - :books: Add `--ask-vault-pass` to cte-dev README

 - Only copy over ssl certificate if the server one is about to expire

### CODE REVIEW

File: templates/var/www/cnx/scripts/settings.js [hunk 1]
```diff
@@ -10,14 +10,12 @@
 
       // Hostname and port for the cnx-archive server
       cnxarchive: {
-        host: '{{ archive_host }}',
-        port: {{ archive_port }}
+        host: '{{ archive_host }}'
```
Comment (pumazi) [inline templates/var/www/cnx/scripts/settings.js]: Maybe `arclishing_domain` now?

**Comment by karenc** [comment]
In addition to the code changes, I added a publishing API key on cte-dev:

```
$ psql -U cnxarchive cnxarchive
psql (9.4.5)
Type "help" for help.

cnxarchive=# \d
cnxarchive=# select * from api_keys;
 id | key | name | groups 
----+-----+------+--------
(0 rows)

cnxarchive=# insert into api_keys (key, name, groups) values (......, 'trusted', '{"g:trusted-publishers"}');
INSERT 0 1
cnxarchive=# \q
```

File: templates/var/www/cnx/scripts/settings.js (hunk 1 above)
Comment (karenc) [inline templates/var/www/cnx/scripts/settings.js]: Yes you're right.

**Comment by karenc** [comment]
Also generated SSL certificate manually using https://letsencrypt.org/:

```
./letsencrypt-auto --manual certonly
```

See http://letsencrypt.readthedocs.org/en/latest/using.html#manual

Then copied `/etc/letsencrypt/live/cte-cnx-dev.cnx.org/fullchain.pem` to `cnx-deploy/environments/cte-dev/secrets/certs/cnx/cnx.crt`

and

`/etc/letsencrypt/live/cte-cnx-dev.cnx.org/privkey.pem` to `cnx-deploy/environments/cte-dev/secrets/certs/cnx/cnx.key`.

Then deploy to cte dev using ansible.

**Comment by karenc** [comment]
^ @pumazi "Only copy over ssl certificate if the server one is about to expire" should allow you to deploy to cte-dev without overwriting the existing lets encrypt SSL certificate.

I'll work on automating the generation of the SSL certificate next.

**Comment by pumazi** [comment]
Should the `openstax_accounts.application_url` values use `arclishing_domain` and `frontend_domain`?

File: templates/etc/cnx/publishing/app.ini [hunk 2]
```diff
@@ -42,7 +42,7 @@ openstax_accounts.application_secret = {{ accounts_consumer_secret }}
 openstax_accounts.groups.moderators =
 # TODO administrators?
 openstax_accounts.groups.administrators =
-openstax_accounts.application_url = http://{{ archive_host }}:{{ archive_port }}/
+openstax_accounts.application_url = https://{{ archive_host }}/
```
Comment (pumazi) [inline templates/etc/cnx/publishing/app.ini]: Should this use `arclishing_domain`?

**Pull request merged**